### DISCUSSION TITLE
git-integration-improvements-in-visual-studio-2017/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#196] [Announcements]
# git-integration-improvements-in-visual-studio-2017/

VS 2017 includes a number of improvements to git integration. In this post I show how you can use git-hooks to validate your commit message format.

https://andrewlock.net/git-integration-improvements-in-visual-studio-2017/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 20 2017, 09:36, in reply to [Nibras Manna](https://disqus.com/by/nibrasmanna/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/196#discussioncomment-6024752)</em>

---

<p>I have prune enabled at a global level in my git config, so VS2017 should honour this (I haven't actually checked yet). To turn this on for your repos use:<br></p><pre><code><br>git config --global fetch.prune true<br></code></pre>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 24 2017, 12:51, in reply to [Ohad Schneider](https://disqus.com/by/ohadschneider/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/196#discussioncomment-6024759)</em>

---

<p>Hmm, that's weird, as you can see in the screenshots, it worked on my machine! There's been a visual studio update or two since then though, so maybe something changed. Or there's an issue with the script you're using of course. </p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 24 2017, 01:01, in reply to [Ohad Schneider](https://disqus.com/by/ohadschneider/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/196#discussioncomment-6024762)</em>

---

<p>Oh I see,  sorry, didn't read that properly! Hopefully that's something they'll improve on soon :) </p>

[a comment by an automated account is not shown]

**Comment by electronum** [comment]
It seems in the commit-msg script  `echo "ERR_MSG" >&2` is missing a `$` to display the actual error message, like this:
```shell
if [[ ! -n "$MATCH_RESULT" ]]; then 
	echo "$ERR_MSG" >&2
	exit 1
fi
```
